modules/ws-client/src/main/scala/trading/client/Update.scala

```diff
@@ -0,0 +1,79 @@
+package trading.client
+
+import trading.domain.{ Alert, Symbol }
+import trading.ws.{ WsIn, WsOut }
+
+import cats.syntax.eq.*
+import io.circe.syntax.*
+
+import tyrian.*
+
+def disconnected(model: Model): (Model, Cmd[Msg]) =
+  model.copy(error = Some("Disconnected from server, please click on Connect.")) -> Cmd.Empty
+
+//TODO:  is there a better way to do this?
+def refocusInput: Cmd[Msg] = Cmd.SideEffect { () =>
+  val elem = org.scalajs.dom.document.querySelector("#symbol-input").asInstanceOf[Tyrian.HTMLInputElement]
+  elem.value = ""
+  elem.focus()
+}
```

Comment:
I saw this and thought "Yikes!" before I noticed the TODO! 😄 
I think I understand what you're trying to do here and why it might be a problem, but ideally, you'd want to model your state in such a way that your view code knew that it needed to give focus back to the field, rather than doing it out of band as you are here.
It's a little difficult for me to grok this without running it / wandering around the code in my editor (so apologies if I'm barking up the wrong tree), but I wonder if it would be possible to include some sort of finite state machine in your model (feels like `sub`, `unsub`, and `error` and leading to that? Maybe?) and use that to know whether the field should have focus?

Reply:
What did the Elm version of this part look like?

Reply:
Yes, this is ugly but I didn't know how to do this better. Here's the ELM code:

```elm
refocusInput : Cmd Msg
refocusInput =
    Task.attempt (\_ -> NoOp) (Dom.focus "symbol-input")
```

I don't know if it's doable to model it with state but I would have to think more about it. I basically want the focus to be on the input box after submit (easy with state) but also after closing the alerts and after clicking unsubscribe in the entries.

Reply:
Oh, no need for apologies, I know very little about all stuff front-end so I might be going wrong about it! 😄 

Cool, will use the `scalajs-dom` directly. I find it a bit of a hack that we need to cast there, though.

Right, the Elm part only does the focus part, and yes, it could be an anti-pattern, I feel it's probably forced 😬. 

Changing the value is something that Elm does for me (see [here](https://github.com/gvolpe/trading/blob/main/web-app/src/View.elm#L29)), so this is one of the key differences I found compared to Tyrian. At least, I was not able to set a `value` attribute to the `input` HTML element, as I have in the Elm app. How is the recommended way of doing updates of a text input box into the model?

Reply:
> I find it a bit of a hack that we need to cast there, though.

I agree, but it is quite a common thing to have to do in scala.js/scalajs-dom - so at least this way it's consistently hacky. ;-)

I imagine that under the covers the Elm Dom package is more or less doing the same thing if it's any consolation... or it's just JS and trying to call focus on whatever it finds that has the right name.

Not being able to set `value := "some value"` due to an ambiguous overload is a bug, I'll raise an issue now. 🤦 

This is the alternative, quite right:

```scala
attribute("value", if model.symbol.show == "XXXXXX" then "" else model.symbol.show)
```

But I'm unclear what's going on with the `XXXXXX`? I'll dig around a bit more...

Reply:
I am stuck with some linking errors at the moment, so unable to test this. Will let you know when I can test it.

Reply:
I managed to inspect the HTML element. I can confirm `attribute("value", model.input)` does not work as a setter.

To get it working as I expect, I had to use `onInput(s => Msg.SymbolChanged(s))` and handle this message accordingly to update the model. Is this how it is supposed to be? That would be a key difference with Elm.

Reply:
OTOH it probably makes sense as I cannot imagine how that can be implemented generically... Definitely something that could use some documentation. If this is the case, I can try to contribute some docs :)

Reply:
https://github.com/PurpleKingdomGames/tyrian/issues/60